applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py

```diff
@@ -376,23 +378,40 @@ def test_compare_TL_UL_2D_triangle(self):
                 ul_dx = ul_mp.Nodes[i].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X)
                 ul_dy = ul_mp.Nodes[i].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y)
 
-                if (ul_dx > 0.0):
-                    self.assertLess((tl_dx - ul_dx) / ul_dx, 1.0e-3)
-                if (ul_dy > 0.0):
-                    self.assertLess((tl_dy - ul_dy) / ul_dy, 1.0e-3)
+                if(abs(tl_dx) > 1e-20):
+                    self.assertLess(abs(tl_dx - ul_dx) / abs(tl_dx), 1.0e-10)
+                if(abs(tl_dy) > 1e-20):
+                    self.assertLess(abs(tl_dy - ul_dy) / abs(tl_dy), 1.0e-10)
 
             # Compare matrices
             for i in range(ul_lhs.Size1()):
                 for j in range(ul_lhs.Size2()):
-                    self.assertLess((ul_lhs[i, j] - tl_lhs[i, j]) / tl_lhs[i, j], 1.0e-3)
+                    self.assertLess(abs(ul_lhs[i, j] - tl_lhs[i, j]) / abs(tl_lhs[i, j]), 1.0e-10)
 
         #self.__post_process(tl_mp)
         #self.__post_process(ul_mp)
 
-    def test_TL_2D_triangle(self):
+    def test_compare_TL_UL_2D_triangle_block(self):
```

Comment:
@loumalouomega @philbucher i changed this starting from #6572 but pusing it further.

one bad thing is that the UL hexa test fails for the level of deformation which is originally used in the test.
The TL test does not fail for the same level of deformation, which implies that we have some problem with the position used as reference in the UL, something that i believe is very related to #6567

Reply:
BTW I obtained this hardcoding in the NR: 

![imagen](https://user-images.githubusercontent.com/19991680/77330059-17dd1500-6d1f-11ea-911f-5be11870faba.png)

Reply:
Honestly i did not doubt that the element got inverted.

the problem is the WHY it gets inverted. One of the checks you do is  that the LHS and RHS are identical (as they should be) for TL and UL. The problem is that the TL test passes and the UL does not ... meaning that at some point the two matrices must be different. no?

Reply:
TL integrates in the undeformed position, therefore these jacobian is constant. If after solving the system the configuration is not broken the problem is solved. UL integrates in the deformed configuration, this means that is not possible to integrate the volume if that configuration is inverted. Maybe removing the error line the result is the same, but this is not the point. You may reach the same solution going few by few without breaking the element.

Reply:
there was an error in my implementation. now it is solved